### PR TITLE
Update Jenkins Tools Kogito node

### PR DESCRIPTION
**Pull request by radtriste**
- Removed useless modules
- Updated module versions
- Updated image name & version

### CODE REVIEW

File: kogito-jenkins-node/modules/maven-installer/3.8.x/module.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+name: org.kie.kogito.maven
+version: "3.8.7"
```
Comment (spolti) [inline kogito-jenkins-node/modules/maven-installer/3.8.x/module.yaml]: 3.8.8?

Comment (radtriste) [reply kogito-jenkins-node/modules/maven-installer/3.8.x/module.yaml]: updated

**Review by spolti — APPROVED** [review]

**Comment by radtriste** [comment]
@spolti Image was built here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/custom/job/tradisso/job/build-kogito-jenkins-node/9/console
and tested here:
https://jenkins-kogito-tools.kogito-cluster-0ad6762cc85bcef5745bb684498c2436-0000.us-south.containers.appdomain.cloud/job/kogito/job/kogito-operator-PR-check/851/

**Comment by radtriste** [comment]
Noe: https://quay.io/repository/kiegroup/kogito-jenkins-node?tab=tags&tag=1.40.0

File: kogito-jenkins-node/modules/maven-installer/3.8.x/module.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+name: org.kie.kogito.maven
+version: "3.8"
+description: "Module used to install maven 3.8"
+envs:
+  - name: "MAVEN_VERSION"
+    value: "3.8.7"
```
Comment (spolti) [inline kogito-jenkins-node/modules/maven-installer/3.8.x/module.yaml]:
```suggestion
    value: "3.8.8"
```

**Pull request merged**